apps/knowledge-base/src/apps/config.tsx

```diff
@@ -0,0 +1,278 @@
+import * as React from 'react';
+import Layout from '~/templates/layout';
+import WelcomeScreen from '~/components/welcome-screen';
+import { useSdk } from '~/providers/sdk-provider';
+import { useNetlify } from '~/providers/netlify-provider';
+import {
+  resetCounters,
+  replaceLocale,
+  getContentTypeSchemaById,
+} from '~/utils/utils';
+import ConfigHeader from '~/components/config-header';
+// import ConfigNetlify from '~/components/config-netlify';
+import ConfigDeploy from '~/components/config-deploy';
+import ConfigSiteSettings from '~/components/config-site-settings';
+import ConfigInstallation from '~/components/config-installation';
+import ConfigDone from '~/components/config-done';
+
+const { useEffect, useState, useMemo } = React;
+
+interface ConfigProps {
+  template: Record<string, any>;
+}
+
+const Config: React.FC<ConfigProps> = (props) => {
+  const netlify = useNetlify();
+  const sdk = useSdk();
+  const defaultLocale = sdk.instance.locales.default;
+  const template = useMemo(
+    () => resetCounters(replaceLocale(props.template, defaultLocale)),
+    [props.template, defaultLocale]
+  );
+  const [isAppInstalled, setIsAppInstalled] = useState(false);
+  const [netlifySelectedSiteUrl, setNetlifySelectedSiteUrl] = useState('');
+  const [netlifySelectedSiteId, setNetlifySelectedSiteId] = useState('');
+  const [
+    netlifySelectedSiteBuildHookUrl,
+    setNetlifySelectedSiteBuildHookUrl,
+  ] = useState('');
+  const [termsAccepted, setTermsAccepted] = useState(false);
+  const [hadDeployed, setHadDeployed] = useState(false);
+
+  useEffect(() => {
+    sdk.instance.app.isInstalled().then(setIsAppInstalled);
+    sdk.instance.app.onConfigure(onSaveOrInstall);
+    sdk.instance.app.onConfigurationCompleted(onFinishSavingOrInstalling);
+  });
+
+  useEffect(() => {
+    loadInitialParams().then(() => sdk.instance.app.setReady());
+  }, []);
+
+  async function loadInitialParams() {
+    const params: Record<string, any> = await sdk.instance.app.getParameters();
+
+    setNetlifySelectedSiteUrl(params?.netlifySelectedSiteUrl);
+    setNetlifySelectedSiteId(params?.netlifySelectedSiteId);
+    setNetlifySelectedSiteBuildHookUrl(params?.netlifySelectedSiteBuildHookUrl);
+    setHadDeployed(!!params?.hadDeployed);
+  }
+
+  async function onFinishSavingOrInstalling() {
+    setIsAppInstalled(true);
+  }
+
+  async function createContentTypes(contentTypes) {
+    const allExistingContentTypes = await sdk.getContentTypes();
```

Comment:
You could use `getCachedContentTypes` to skip networking for this one. `getContentTypes` makes sense after updates to CTs were made by an app itself.